File: C#/Searching.cs
using System;

namespace LB1
{
    public class Searching
    {
        private readonly int idThread;
        private int start;
        private int end;
        private Mass mass;
        private MinCollector collector;

        public Searching(int id, int start, int end, Mass mass, MinCollector collector)
        {
            idThread = id;
            this.start = start;
            this.end = end;
            this.mass = mass;
            this.collector = collector;
        }

        public void Run()
        {
            int[] minElement = mass.GetMinElement(start, end);

            Console.WriteLine($"In thread {idThread} min element was found {minElement[0]} which index {minElement[1]}");

            collector.Collect(minElement[0], minElement[1]);
        }
    }
}

File: Java/Searching.java
public class Searching extends Thread {
    private final int startIndex;
    private final int endIndex;
    private final Mass mass;
    private final MinCollector collector;

    public Searching(int startIndex, int endIndex, Mass mass, MinCollector collector) {
        this.startIndex = startIndex;
        this.endIndex = endIndex;
        this.mass = mass;
        this.collector = collector;
    }

    @Override
    public void run() {
        int[] minElement = mass.GetMinElement(startIndex, endIndex);
        System.out.println("In thread " + this.getName() + " min element was found " + minElement[0] + " which index " + minElement[1]);

        collector.collect(minElement[0], minElement[1]);
        collector.incThreadCount();
    }
}
